include/capstone/tricore.h

```diff
@@ -18,40 +18,43 @@ extern "C" {
 #pragma warning(disable : 4201)
 #endif
 
-//> Operand type for instruction's operands
+/// Operand type for instruction's operands
 typedef enum tricore_op_type {
-	TRICORE_OP_INVALID = 0, // = CS_OP_INVALID (Uninitialized).
-	TRICORE_OP_REG, // = CS_OP_REG (Register operand).
-	TRICORE_OP_IMM, // = CS_OP_IMM (Immediate operand).
-	TRICORE_OP_MEM, // = CS_OP_MEM (Memory operand).
+	TRICORE_OP_INVALID = 0, ///< CS_OP_INVALID (Uninitialized).
+	TRICORE_OP_REG, ///< CS_OP_REG (Register operand).
+	TRICORE_OP_IMM, ///< CS_OP_IMM (Immediate operand).
+	TRICORE_OP_MEM, ///< CS_OP_MEM (Memory operand).
 } tricore_op_type;
 
-// Instruction's operand referring to memory
-// This is associated with TRICORE_OP_MEM operand type above
+/// Instruction's operand referring to memory
+/// This is associated with TRICORE_OP_MEM operand type above
 typedef struct tricore_op_mem {
-	uint8_t base; // base register
-	int32_t disp; // displacement/offset value
+	uint8_t base; ///< base register
+	int32_t disp; ///< displacement/offset value
 } tricore_op_mem;
 
-// Instruction operand
+/// Instruction operand
 typedef struct cs_tricore_op {
-	tricore_op_type type; // operand type
+	tricore_op_type type; ///< operand type
 	union {
-		unsigned int reg; // register value for REG operand
-		int32_t imm; // immediate value for IMM operand
-		tricore_op_mem mem; // base/disp value for MEM operand
+		unsigned int reg; ///< register value for REG operand
+		int32_t imm; ///< immediate value for IMM operand
+		tricore_op_mem mem; ///< base/disp value for MEM operand
 	};
```

Comment:
Took me until here to understand how you fill memory operands :D
Doing it via `fill_mem()` is fine.

Nonetheless, please set the `TRICORE_OP_MEM = CS_OP_MEM` and update the operand type of the base register and disponent with `TRICORE_OP_MEM` (in `fill_tricore_imm()` via `tricore->operands[tricore->op_count].type |= TRICORE_OP_MEM`).

If we extend TriCore's `td` files in the future and define complex memory operands there (memory operands which have the base register and disponent as sub-operands defined), it is easier to modify the code here.